In [1]:
import numpy as np 
import pandas as pd 

In [2]:
import json 
from sklearn import preprocessing

In [3]:
import matplotlib.pyplot as plt

In [4]:
import math as m 

In [5]:
%load_ext ipython_memory_usage

Enabling IPython Memory Usage, use %imu_start to begin, %imu_stop to end


In [6]:
%imu_start 

'IPython Memory Usage started'

In [6] used 0.4 MiB RAM in 0.12s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 150.6 MiB


### EDA de la dataFrame 'Athlete_events'

In [7]:
df_athlete=pd.read_csv('data/athlete_events.csv',sep=',')

df_athlete

Unnamed: 0                Name Sex   Age  Height      Team  NOC  \
0                0     AALERUD Katrine   F  27.0     NaN    Norway  NOR   
1                1         ABAD Nestor   M  28.0   165.0     Spain  ESP   
2                2   ABAGNALE Giovanni   M  26.0   198.0     Italy  ITA   
3                3      ABALDE Alberto   M  26.0   200.0     Spain  ESP   
4                4       ABALDE Tamara   F  32.0   192.0     Spain  ESP   
...            ...                 ...  ..   ...     ...       ...  ...   
282985      271111          Andrzej ya   M  29.0   179.0  Poland-1  POL   
282986      271112            Piotr ya   M  27.0   176.0    Poland  POL   
282987      271113            Piotr ya   M  27.0   176.0    Poland  POL   
282988      271114  Tomasz Ireneusz ya   M  30.0   185.0    Poland  POL   
282989      271115  Tomasz Ireneusz ya   M  34.0   185.0    Poland  POL   

              Games  Year  Season            City                Sport Medal  \
0        Tokyo 2020  2020  Summer           Tokyo         Cycling Road   NaN   
1        Tokyo 2020  2020  Summer           Tokyo  Artistic Gymnastics   NaN   
2        Tokyo 2020  2020  Summer           Tokyo               Rowing   NaN   
3        Tokyo 2020  2020  Summer           Tokyo           Basketball   NaN   
4        Tokyo 2020  2020  Summer           Tokyo           Basketball   NaN   
...             ...   ...     ...             ...                  ...   ...   
282985  1976 Winter  1976  Winter       Innsbruck                 Luge   NaN   
282986  2014 Winter  2014  Winter           Sochi          Ski Jumping   NaN   
282987  2014 Winter  2014  Winter           Sochi          Ski Jumping   NaN   
282988  1998 Winter  1998  Winter          Nagano            Bobsleigh   NaN   
282989  2002 Winter  2002  Winter  Salt Lake City            Bobsleigh   NaN   

                                           Event        ID  Weight  \
0                                            NaN       NaN     NaN   
1                                            NaN       NaN     NaN   
2                                            NaN       NaN     NaN   
3                                            NaN       NaN     NaN   
4                                            NaN       NaN     NaN   
...                                          ...       ...     ...   
282985                Luge Mixed (Men)'s Doubles  135569.0    89.0   
282986  Ski Jumping Men's Large Hill, Individual  135570.0    59.0   
282987        Ski Jumping Men's Large Hill, Team  135570.0    59.0   
282988                      Bobsleigh Men's Four  135571.0    96.0   
282989                      Bobsleigh Men's Four  135571.0    96.0   

                     Country_org  
0                          Japan  
1                          Japan  
2                          Japan  
3                          Japan  
4                          Japan  
...                          ...  
282985                    Canada  
282986        Russian Federation  
282987        Russian Federation  
282988                     Japan  
282989  United States of America  

[282990 rows x 17 columns]

In [7] used 189.8 MiB RAM in 0.68s (system mean cpu 25%, single max cpu 92%), peaked 0.0 MiB above final usage, current RAM usage now 340.4 MiB


### DataFrame dictionnary

In [8]:
df_dictionnary=pd.read_csv('data/dictionary.csv')

df_dictionnary=df_dictionnary.rename(columns={'GDP per Capita':'PIB/habitant','Code':'NOC'})

df_dictionnary.isna().sum()

Country          0
NOC              0
Population       5
PIB/habitant    25
dtype: int64

In [8] used -3.9 MiB RAM in 0.11s (system mean cpu 28%, single max cpu 58%), peaked 0.0 MiB above final usage, current RAM usage now 336.6 MiB


### On cherche les athlètes qui n'ont pas de team et ceux des îles TUVALU

In [9]:
df_athlete[df_athlete['NOC']=='UNK']


Unnamed: 0          Name Sex  Age  Height     Team  NOC        Games  \
72954        61080  Fritz Eccard   M  NaN     NaN  Unknown  UNK  1912 Summer   
142595      130721     A. Laffen   M  NaN     NaN  Unknown  UNK  1912 Summer   

        Year  Season       City             Sport Medal  \
72954   1912  Summer  Stockholm  Art Competitions   NaN   
142595  1912  Summer  Stockholm  Art Competitions   NaN   

                                      Event       ID  Weight Country_org  
72954   Art Competitions Mixed Architecture  31292.0     NaN      Sweden  
142595  Art Competitions Mixed Architecture  65813.0     NaN      Sweden

In [9] used 0.0 MiB RAM in 0.14s (system mean cpu 20%, single max cpu 62%), peaked 0.0 MiB above final usage, current RAM usage now 336.6 MiB


Fritez Eccard a participé dans l'équipe Suédoise et A.Laffen .... aucunes traces je sais pas qui est ce mec 

In [10]:
df_athlete[df_athlete['NOC']=='ROT'].nunique()


Unnamed: 0     12
Name           10
Sex             2
Age             7
Height          7
Team            1
NOC             1
Games           1
Year            1
Season          1
City            1
Sport           3
Medal           0
Event          12
ID             10
Weight          9
Country_org     1
dtype: int64

In [10] used 0.5 MiB RAM in 0.14s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 337.1 MiB


12 participations parmi 10 athlètes qui ont participé aux JO ces dernières années sous la team des athlètes réfugiés. 

In [11]:
df_athlete=df_athlete.query(" NOC !='ROC' and NOC != 'UNK' ")

In [11] used 5.2 MiB RAM in 0.16s (system mean cpu 23%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 342.3 MiB


## Base données région 

In [12]:
df_regions=pd.read_csv('data/noc_regions.csv')

df_regions=df_regions.rename(columns={'region':'Country'})
df_regions

NOC      Country                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

In [12] used -7.6 MiB RAM in 0.11s (system mean cpu 22%, single max cpu 100%), peaked 7.6 MiB above final usage, current RAM usage now 334.7 MiB


In [13]:
df_regions.nunique()

NOC        230
Country    206
notes       21
dtype: int64

In [13] used -4.5 MiB RAM in 0.11s (system mean cpu 11%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 330.2 MiB


Il y a des régions qui ont le même NOC 

In [14]:
duplicates=[df_regions.duplicated(subset='Country')]

In [14] used 0.1 MiB RAM in 0.11s (system mean cpu 20%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 330.3 MiB


In [15]:
df_regions[df_regions.duplicated(subset='Country')]

NOC         Country                   notes
12   AUS       Australia                     NaN
54   CZE  Czech Republic                     NaN
76   GDR         Germany                     NaN
79   GER         Germany                     NaN
81   GRE          Greece                     NaN
88   HKG           China               Hong Kong
126  MAS        Malaysia                     NaN
143  NBO        Malaysia            North Borneo
147  NFL          Canada            Newfoundland
171  RUS          Russia                     NaN
173  SAA         Germany                     NaN
185  SRB          Serbia                     NaN
197  TCH  Czech Republic                     NaN
208  TUV             NaN                  Tuvalu
210  UAR           Syria    United Arab Republic
213  UNK             NaN                 Unknown
214  URS          Russia                     NaN
222  VNM         Vietnam                     NaN
223  WIF        Trinidad  West Indies Federation
225  YEM           Yemen                     NaN
226  YMD           Yemen             South Yemen
227  YUG          Serbia              Yugoslavia
229  ZIM        Zimbabwe                     NaN

In [15] used 0.1 MiB RAM in 0.13s (system mean cpu 27%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 330.4 MiB


In [16]:
df_regions.isnull().sum()

NOC          0
Country      3
notes      209
dtype: int64

In [16] used 0.0 MiB RAM in 0.11s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 330.4 MiB


In [17]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [17] used -2.1 MiB RAM in 0.11s (system mean cpu 20%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 328.3 MiB


Il y a  3 des regions qui ne sont pas attribuées, ROT ( refugee olympic team)  , TUVALU ( 9 archipels qui constituent un État ) et UNK qui correspond à Unknown 

### On ajoute les iles TUVALU 

In [18]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [18] used 0.0 MiB RAM in 0.11s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 328.3 MiB


In [19]:
df_regions=df_regions.query(" NOC != 'ROT' and NOC != 'UNK' ")

df_regions.isnull().sum()

NOC          0
Country      1
notes      209
dtype: int64

In [19] used -13.3 MiB RAM in 0.11s (system mean cpu 13%, single max cpu 100%), peaked 13.3 MiB above final usage, current RAM usage now 315.0 MiB


In [20]:
df_regions['Country']=df_regions['Country'].fillna('Tuvalu')

In [20] used 0.1 MiB RAM in 0.11s (system mean cpu 21%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 315.1 MiB


<ipython-input-20-0f7f308fe64b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_regions['Country']=df_regions['Country'].fillna('Tuvalu')


In [21]:
df_regions[df_regions['Country']=='Tuvalu']

NOC Country   notes
208  TUV  Tuvalu  Tuvalu

In [21] used -4.6 MiB RAM in 0.11s (system mean cpu 15%, single max cpu 100%), peaked 2.0 MiB above final usage, current RAM usage now 310.6 MiB


In [22]:
df_regions.isnull().sum()

NOC          0
Country      0
notes      209
dtype: int64

In [22] used 0.0 MiB RAM in 0.11s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 310.6 MiB


# On va merge les dataframes 

## Merge en left 

In [23]:
df_complete=pd.merge(pd.merge(df_athlete,df_regions,on='NOC',how='left'),df_dictionnary,on='NOC',how='left')
#df_complete=df_complete.drop(labels='',axis=1)

df_complete

Unnamed: 0                Name Sex   Age  Height      Team  NOC  \
0                0     AALERUD Katrine   F  27.0     NaN    Norway  NOR   
1                1         ABAD Nestor   M  28.0   165.0     Spain  ESP   
2                2   ABAGNALE Giovanni   M  26.0   198.0     Italy  ITA   
3                3      ABALDE Alberto   M  26.0   200.0     Spain  ESP   
4                4       ABALDE Tamara   F  32.0   192.0     Spain  ESP   
...            ...                 ...  ..   ...     ...       ...  ...   
282622      271111          Andrzej ya   M  29.0   179.0  Poland-1  POL   
282623      271112            Piotr ya   M  27.0   176.0    Poland  POL   
282624      271113            Piotr ya   M  27.0   176.0    Poland  POL   
282625      271114  Tomasz Ireneusz ya   M  30.0   185.0    Poland  POL   
282626      271115  Tomasz Ireneusz ya   M  34.0   185.0    Poland  POL   

              Games  Year  Season  ... Medal  \
0        Tokyo 2020  2020  Summer  ...   NaN   
1        Tokyo 2020  2020  Summer  ...   NaN   
2        Tokyo 2020  2020  Summer  ...   NaN   
3        Tokyo 2020  2020  Summer  ...   NaN   
4        Tokyo 2020  2020  Summer  ...   NaN   
...             ...   ...     ...  ...   ...   
282622  1976 Winter  1976  Winter  ...   NaN   
282623  2014 Winter  2014  Winter  ...   NaN   
282624  2014 Winter  2014  Winter  ...   NaN   
282625  1998 Winter  1998  Winter  ...   NaN   
282626  2002 Winter  2002  Winter  ...   NaN   

                                           Event        ID Weight  \
0                                            NaN       NaN    NaN   
1                                            NaN       NaN    NaN   
2                                            NaN       NaN    NaN   
3                                            NaN       NaN    NaN   
4                                            NaN       NaN    NaN   
...                                          ...       ...    ...   
282622                Luge Mixed (Men)'s Doubles  135569.0   89.0   
282623  Ski Jumping Men's Large Hill, Individual  135570.0   59.0   
282624        Ski Jumping Men's Large Hill, Team  135570.0   59.0   
282625                      Bobsleigh Men's Four  135571.0   96.0   
282626                      Bobsleigh Men's Four  135571.0   96.0   

                     Country_org  Country_x notes Country_y  Population  \
0                          Japan     Norway   NaN    Norway   5195921.0   
1                          Japan      Spain   NaN     Spain  46418269.0   
2                          Japan      Italy   NaN     Italy  60802085.0   
3                          Japan      Spain   NaN     Spain  46418269.0   
4                          Japan      Spain   NaN     Spain  46418269.0   
...                          ...        ...   ...       ...         ...   
282622                    Canada     Poland   NaN    Poland  37999494.0   
282623        Russian Federation     Poland   NaN    Poland  37999494.0   
282624        Russian Federation     Poland   NaN    Poland  37999494.0   
282625                     Japan     Poland   NaN    Poland  37999494.0   
282626  United States of America     Poland   NaN    Poland  37999494.0   

        PIB/habitant  
0       74400.369777  
1       25831.582305  
2       29957.804315  
3       25831.582305  
4       25831.582305  
...              ...  
282622  12554.547554  
282623  12554.547554  
282624  12554.547554  
282625  12554.547554  
282626  12554.547554  

[282627 rows x 22 columns]

In [23] used 96.6 MiB RAM in 0.35s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 407.2 MiB


In [24]:
df_complete=df_complete.query(" NOC !='ROC' ")

In [24] used -9.4 MiB RAM in 0.16s (system mean cpu 9%, single max cpu 100%), peaked 11.6 MiB above final usage, current RAM usage now 397.8 MiB


In [25]:
#df_complete.info()

In [25] used 0.0 MiB RAM in 0.11s (system mean cpu 9%, single max cpu 67%), peaked 0.0 MiB above final usage, current RAM usage now 397.8 MiB


282 627 lignes , c'est normal on a enlevé les  réfugiés

In [26]:
df_complete.loc[(df_complete['Medal'].isna(),'Medal')]='Pas_medaille'



In [26] used 0.8 MiB RAM in 0.13s (system mean cpu 19%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 398.6 MiB


## Encoding test 

In [27]:
dictio_score_medaille={'Pas_medaille':0,
                      'Bronze':1,
                      'Silver':2,
                      'Gold':3}

df_complete['Score_medaille']=df_complete['Medal'].map(dictio_score_medaille)

df_complete

Unnamed: 0                Name Sex   Age  Height      Team  NOC  \
0                0     AALERUD Katrine   F  27.0     NaN    Norway  NOR   
1                1         ABAD Nestor   M  28.0   165.0     Spain  ESP   
2                2   ABAGNALE Giovanni   M  26.0   198.0     Italy  ITA   
3                3      ABALDE Alberto   M  26.0   200.0     Spain  ESP   
4                4       ABALDE Tamara   F  32.0   192.0     Spain  ESP   
...            ...                 ...  ..   ...     ...       ...  ...   
282622      271111          Andrzej ya   M  29.0   179.0  Poland-1  POL   
282623      271112            Piotr ya   M  27.0   176.0    Poland  POL   
282624      271113            Piotr ya   M  27.0   176.0    Poland  POL   
282625      271114  Tomasz Ireneusz ya   M  30.0   185.0    Poland  POL   
282626      271115  Tomasz Ireneusz ya   M  34.0   185.0    Poland  POL   

              Games  Year  Season  ...  \
0        Tokyo 2020  2020  Summer  ...   
1        Tokyo 2020  2020  Summer  ...   
2        Tokyo 2020  2020  Summer  ...   
3        Tokyo 2020  2020  Summer  ...   
4        Tokyo 2020  2020  Summer  ...   
...             ...   ...     ...  ...   
282622  1976 Winter  1976  Winter  ...   
282623  2014 Winter  2014  Winter  ...   
282624  2014 Winter  2014  Winter  ...   
282625  1998 Winter  1998  Winter  ...   
282626  2002 Winter  2002  Winter  ...   

                                           Event        ID Weight  \
0                                            NaN       NaN    NaN   
1                                            NaN       NaN    NaN   
2                                            NaN       NaN    NaN   
3                                            NaN       NaN    NaN   
4                                            NaN       NaN    NaN   
...                                          ...       ...    ...   
282622                Luge Mixed (Men)'s Doubles  135569.0   89.0   
282623  Ski Jumping Men's Large Hill, Individual  135570.0   59.0   
282624        Ski Jumping Men's Large Hill, Team  135570.0   59.0   
282625                      Bobsleigh Men's Four  135571.0   96.0   
282626                      Bobsleigh Men's Four  135571.0   96.0   

                     Country_org  Country_x  notes Country_y  Population  \
0                          Japan     Norway    NaN    Norway   5195921.0   
1                          Japan      Spain    NaN     Spain  46418269.0   
2                          Japan      Italy    NaN     Italy  60802085.0   
3                          Japan      Spain    NaN     Spain  46418269.0   
4                          Japan      Spain    NaN     Spain  46418269.0   
...                          ...        ...    ...       ...         ...   
282622                    Canada     Poland    NaN    Poland  37999494.0   
282623        Russian Federation     Poland    NaN    Poland  37999494.0   
282624        Russian Federation     Poland    NaN    Poland  37999494.0   
282625                     Japan     Poland    NaN    Poland  37999494.0   
282626  United States of America     Poland    NaN    Poland  37999494.0   

        PIB/habitant Score_medaille  
0       74400.369777              0  
1       25831.582305              0  
2       29957.804315              0  
3       25831.582305              0  
4       25831.582305              0  
...              ...            ...  
282622  12554.547554              0  
282623  12554.547554              0  
282624  12554.547554              0  
282625  12554.547554              0  
282626  12554.547554              0  

[282627 rows x 23 columns]

In [27] used 18.3 MiB RAM in 0.25s (system mean cpu 13%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 417.0 MiB


In [28]:
#Marche aussi 
''' 
medaille_categories=['Pas_medaille','Bronze','Silver','Gold']
encode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])

df_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])

df_complete
''' 

" \nmedaille_categories=['Pas_medaille','Bronze','Silver','Gold']\nencode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])\n\ndf_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])\n\ndf_complete\n"

In [28] used -0.0 MiB RAM in 0.11s (system mean cpu 28%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 416.9 MiB


## Analyse valeurs nulles 

In [29]:
df_complete.isna().sum()

Unnamed: 0             0
Name                   0
Sex                  156
Age                 9628
Height             67142
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Medal                  0
Event               9278
ID                 11513
Weight             74386
Country_org         1733
Country_x            419
notes             277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
dtype: int64

In [29] used 2.3 MiB RAM in 0.26s (system mean cpu 4%, single max cpu 64%), peaked 0.0 MiB above final usage, current RAM usage now 419.2 MiB


156 sports où on sait pas ce que c'est .... et des country (26 221) qui ne sont pas attribués non plus 

Pour les Country , on va merge 

In [30]:
#df_complete_test=df_complete.copy()

df_complete=pd.merge(df_complete,df_regions,on='NOC',how='left')

df_complete['Country']=df_complete['Country_y'].fillna(df_complete['Country_x'])

In [30] used 37.0 MiB RAM in 0.19s (system mean cpu 5%, single max cpu 50%), peaked 0.0 MiB above final usage, current RAM usage now 456.2 MiB


In [31]:
df_complete= df_complete.drop(['Country_x', 'Country_y'], axis=1)

In [31] used 12.6 MiB RAM in 0.16s (system mean cpu 29%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 468.8 MiB


In [32]:
liste_pays=df_regions['Country'].tolist()
liste_NOC=df_regions['NOC'].tolist()

dictio_NOC_pays=dict(zip(liste_NOC,liste_pays))

dictio_NOC_pays['SGP']='Singapour'
dictio_NOC_pays['LBN']='Liban'


df_complete['Country']=df_complete['Country'].fillna(df_complete['NOC'].map(dictio_NOC_pays))
#liste_region

#df_complete_test.isna().sum()

In [32] used 0.3 MiB RAM in 0.15s (system mean cpu 19%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 469.1 MiB


In [33]:
#test=df_complete_test[df_complete_test['Country'].isna()]

#test['Country']=test['Country'].fillna(test['NOC'].map(dictio_NOC_pays))

#test.groupby('NOC')['Name'].count().reset_index()

In [33] used -15.4 MiB RAM in 0.11s (system mean cpu 20%, single max cpu 67%), peaked 15.4 MiB above final usage, current RAM usage now 453.7 MiB


Il y a quatre NOC qui ne correspondent à aucun pays dans notre EOR : réfugiés , ROT réfugiés aussi , LBN c'est le Liban (Lebanon en anglais )

In [34]:
df_complete=df_complete.query(" NOC !='ROT' and NOC !='EOR' ")



In [34] used -25.7 MiB RAM in 0.19s (system mean cpu 18%, single max cpu 80%), peaked 25.7 MiB above final usage, current RAM usage now 428.0 MiB


In [35]:
df_complete=df_complete.drop(['notes_x','notes_y'],axis=1)

In [35] used 0.1 MiB RAM in 0.15s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 428.0 MiB


## Opérations sur les sports 

In [36]:
liste_sports=df_complete['Sport'].unique().tolist()

print(liste_sports)

['Cycling Road', 'Artistic Gymnastics', 'Rowing', 'Basketball', 'Swimming', 'Karate', 'Wrestling', 'Handball', 'Rhythmic Gymnastics', 'Athletics', 'Judo', 'Shooting', 'Table Tennis', 'Football', 'Taekwondo', 'Fencing', 'Badminton', 'Boxing', 'Archery', 'Beach Volleyball', 'Sailing', 'Weightlifting', 'Hockey', 'Trampoline Gymnastics', 'Triathlon', 'Canoe Slalom', 'Water Polo', 'Surfing', 'Canoe Sprint', 'Rugby Sevens', 'Volleyball', nan, 'Equestrian', 'Artistic Swimming', 'Cycling Track', 'Golf', 'Skateboarding', 'Marathon Swimming', 'Cycling BMX Racing', 'Tennis', 'Baseball/Softball', 'Diving', 'Cycling Mountain Bike', '3x3 Basketball', 'Modern Pentathlon', 'Cycling BMX Freestyle', 'Sport Climbing', 'Tug-Of-War', 'Speed Skating', 'Cross Country Skiing', 'Ice Hockey', 'Biathlon', 'Gymnastics', 'Art Competitions', 'Alpine Skiing', 'Luge', 'Bobsleigh', 'Equestrianism', 'Cycling', 'Canoeing', 'Figure Skating', 'Softball', 'Synchronized Swimming', 'Nordic Combined', 'Baseball', 'Freestyle S

In [37]:
table_sport=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sport

Sport  Name
0   3x3 Basketball    56
1      Aeronautics     1
2    Alpine Skiing  8829
3         Alpinism    25
4          Archery  2461
..             ...   ...
79      Tug-Of-War   170
80      Volleyball  3666
81      Water Polo  4119
82   Weightlifting  4128
83       Wrestling  7425

[84 rows x 2 columns]

In [37] used 1.1 MiB RAM in 0.17s (system mean cpu 17%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 429.1 MiB


In [38]:
df_complete=df_complete[df_complete['Year']>=2000]

In [38] used 3.0 MiB RAM in 0.13s (system mean cpu 13%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 432.1 MiB


In [39]:
df_complete

Unnamed: 0                          Name Sex   Age  Height     Team  \
0                0               AALERUD Katrine   F  27.0     NaN   Norway   
1                1                   ABAD Nestor   M  28.0   165.0    Spain   
2                2             ABAGNALE Giovanni   M  26.0   198.0    Italy   
3                3                ABALDE Alberto   M  26.0   200.0    Spain   
4                4                 ABALDE Tamara   F  32.0   192.0    Spain   
...            ...                           ...  ..   ...     ...      ...   
282620      271109  Aleksandr Viktorovich Zyuzin   M  28.0   183.0   Russia   
282621      271110        Olga Igorevna Zyuzkova   F  33.0   171.0  Belarus   
282623      271112                      Piotr ya   M  27.0   176.0   Poland   
282624      271113                      Piotr ya   M  27.0   176.0   Poland   
282626      271115            Tomasz Ireneusz ya   M  34.0   185.0   Poland   

        NOC        Games  Year  Season  ...                Sport  \
0       NOR   Tokyo 2020  2020  Summer  ...         Cycling Road   
1       ESP   Tokyo 2020  2020  Summer  ...  Artistic Gymnastics   
2       ITA   Tokyo 2020  2020  Summer  ...               Rowing   
3       ESP   Tokyo 2020  2020  Summer  ...           Basketball   
4       ESP   Tokyo 2020  2020  Summer  ...           Basketball   
...     ...          ...   ...     ...  ...                  ...   
282620  RUS  2004 Summer  2004  Summer  ...               Rowing   
282621  BLR  2016 Summer  2016  Summer  ...           Basketball   
282623  POL  2014 Winter  2014  Winter  ...          Ski Jumping   
282624  POL  2014 Winter  2014  Winter  ...          Ski Jumping   
282626  POL  2002 Winter  2002  Winter  ...            Bobsleigh   

               Medal                                     Event        ID  \
0       Pas_medaille                                       NaN       NaN   
1       Pas_medaille                                       NaN       NaN   
2       Pas_medaille                                       NaN       NaN   
3       Pas_medaille                                       NaN       NaN   
4       Pas_medaille                                       NaN       NaN   
...              ...                                       ...       ...   
282620  Pas_medaille    Rowing Men's Lightweight Coxless Fours  135567.0   
282621  Pas_medaille             Basketball Women's Basketball  135568.0   
282623  Pas_medaille  Ski Jumping Men's Large Hill, Individual  135570.0   
282624  Pas_medaille        Ski Jumping Men's Large Hill, Team  135570.0   
282626  Pas_medaille                      Bobsleigh Men's Four  135571.0   

        Weight               Country_org   Population  PIB/habitant  \
0          NaN                     Japan    5195921.0  74400.369777   
1          NaN                     Japan   46418269.0  25831.582305   
2          NaN                     Japan   60802085.0  29957.804315   
3          NaN                     Japan   46418269.0  25831.582305   
4          NaN                     Japan   46418269.0  25831.582305   
...        ...                       ...          ...           ...   
282620    72.0                    Greece  144096812.0   9092.580536   
282621    69.0                    Brazil    9513000.0   5740.456495   
282623    59.0        Russian Federation   37999494.0  12554.547554   
282624    59.0        Russian Federation   37999494.0  12554.547554   
282626    96.0  United States of America   37999494.0  12554.547554   

        Score_medaille  Country  
0                    0   Norway  
1                    0    Spain  
2                    0    Italy  
3                    0    Spain  
4                    0    Spain  
...                ...      ...  
282620               0   Russia  
282621               0  Belarus  
282623               0   Poland  
282624               0   Poland  
282626               0   Poland  

[96730 rows x 21 columns]

In [39] used 10.8 MiB RAM in 0.18s (system mean cpu 13%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 442.9 MiB


## Opérations sur les Ages

In [40]:
table_sport_age=df_complete.groupby('Sport')['Age'].mean().apply(m.ceil).reset_index()

table_sport_age

Sport  Age
0        3x3 Basketball   30
1         Alpine Skiing   25
2               Archery   27
3   Artistic Gymnastics   25
4     Artistic Swimming   24
..                  ...  ...
64            Triathlon   29
65           Volleyball   27
66           Water Polo   27
67        Weightlifting   25
68            Wrestling   27

[69 rows x 2 columns]

In [40] used 0.1 MiB RAM in 0.13s (system mean cpu 21%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 443.0 MiB


In [41]:
liste_sport=table_sport_age['Sport'].tolist()
liste_age=table_sport_age['Age'].tolist()

dictio_age_sport=dict(zip(liste_sport,liste_age))

In [41] used 0.0 MiB RAM in 0.11s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 443.1 MiB


In [42]:
df_complete['Age']=df_complete['Age'].fillna(df_complete['Sport'].map(dictio_age_sport))

df_complete.isna().sum()

Unnamed: 0            0
Name                  0
Sex                 156
Age                 156
Height             7658
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport               156
Medal                 0
Event              9249
ID                11484
Weight            12694
Country_org           0
Population         2855
PIB/habitant       5741
Score_medaille        0
Country               0
dtype: int64

In [42] used 1.3 MiB RAM in 0.19s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 444.4 MiB


In [43]:
df_complete[df_complete['Age'].isna()].groupby('Medal')['Name'].count().reset_index()

Medal  Name
0  Pas_medaille   156

In [43] used 0.0 MiB RAM in 0.11s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 444.5 MiB


In [44]:
#df_tokyo=pd.read_csv('data/athlete_events_tokyo_cleaned.csv')

#df_tokyo.isna().sum()

In [44] used 0.0 MiB RAM in 0.10s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 444.5 MiB


Enfaite il y a 156 personnes qui manquent prcq c'est les qu'on a pas les sport 

In [45]:
df_complete=df_complete.dropna(subset='Age')

In [45] used 0.4 MiB RAM in 0.13s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 444.9 MiB


In [46]:
df_complete.isna().sum()

Unnamed: 0            0
Name                  0
Sex                   0
Age                   0
Height             7502
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Medal                 0
Event              9093
ID                11328
Weight            12538
Country_org           0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [46] used 0.4 MiB RAM in 0.19s (system mean cpu 7%, single max cpu 67%), peaked 0.0 MiB above final usage, current RAM usage now 445.3 MiB


## Opérations sur les poids et les tailles 

In [47]:
df_complete.isna().sum()

Unnamed: 0            0
Name                  0
Sex                   0
Age                   0
Height             7502
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Medal                 0
Event              9093
ID                11328
Weight            12538
Country_org           0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [47] used 0.4 MiB RAM in 0.17s (system mean cpu 6%, single max cpu 71%), peaked 0.0 MiB above final usage, current RAM usage now 445.7 MiB


In [48]:
table_taille=df_complete[df_complete['Height'].isna()==False].groupby(['Sport'])['Height'].mean().apply(m.ceil).reset_index()

liste_sport=table_taille['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_taille=table_taille['Height'].tolist()
print(table_taille)
liste_sport_sex=[]
for i in range(len(liste_sport)):
    liste_sport_sex.append((liste_sport[i],liste_taille[i]))
#print(liste_sport_sex)

dictio_sport_sex=dict(zip(liste_sport,liste_taille))



                  Sport  Height
0        3x3 Basketball     188
1         Alpine Skiing     176
2               Archery     174
3   Artistic Gymnastics     162
4     Artistic Swimming     172
..                  ...     ...
62            Triathlon     174
63           Volleyball     190
64           Water Polo     186
65        Weightlifting     167
66            Wrestling     174

[67 rows x 2 columns]
In [48] used 2.2 MiB RAM in 0.14s (system mean cpu 6%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 447.9 MiB


In [49]:
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


df_complete.isna().sum()

<ipython-input-49-30982cdfd39c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


Unnamed: 0            0
Name                  0
Sex                   0
Age                   0
Height              159
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Medal                 0
Event              9093
ID                11328
Weight            12538
Country_org           0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [49] used 0.1 MiB RAM in 0.17s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 447.9 MiB


In [50]:
#df_complete[df_complete['Height'].isna()]

In [50] used 0.0 MiB RAM in 0.11s (system mean cpu 18%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 447.9 MiB


In [51]:
table_poids=df_complete[df_complete['Weight'].isna()==False].groupby(['Sport'])['Weight'].mean().apply(m.ceil).reset_index()

liste_sport_2=table_poids['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids=table_poids['Weight'].tolist()



dictio_sport_poids=dict(zip(liste_sport_2,liste_poids))

In [51] used 10.5 MiB RAM in 0.15s (system mean cpu 28%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 458.4 MiB


In [52]:
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


df_complete.info()

<class 'pandas.core.frame.DataFrame'>
Index: 96574 entries, 0 to 282626
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      96574 non-null  int64  
 1   Name            96574 non-null  object 
 2   Sex             96574 non-null  object 
 3   Age             96574 non-null  float64
 4   Height          96415 non-null  float64
 5   Team            96574 non-null  object 
 6   NOC             96574 non-null  object 
 7   Games           96574 non-null  object 
 8   Year            96574 non-null  int64  
 9   Season          96574 non-null  object 
 10  City            96574 non-null  object 
 11  Sport           96574 non-null  object 
 12  Medal           96574 non-null  object 
 13  Event           87481 non-null  object 
 14  ID              85246 non-null  float64
 15  Weight          94584 non-null  float64
 16  Country_org     96574 non-null  object 
 17  Population      93728 non-null  flo

<ipython-input-52-f91662992dfc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


In [53]:
df_complete=df_complete[df_complete['Season']=='Summer']

df_complete

Unnamed: 0                          Name Sex   Age  Height       Team  \
0                0               AALERUD Katrine   F  27.0   175.0     Norway   
1                1                   ABAD Nestor   M  28.0   165.0      Spain   
2                2             ABAGNALE Giovanni   M  26.0   198.0      Italy   
3                3                ABALDE Alberto   M  26.0   200.0      Spain   
4                4                 ABALDE Tamara   F  32.0   192.0      Spain   
...            ...                           ...  ..   ...     ...        ...   
282616      271105      Fernando scar Zylberberg   M  23.0   168.0  Argentina   
282617      271106      Fernando scar Zylberberg   M  27.0   168.0  Argentina   
282619      271108  Aleksandr Viktorovich Zyuzin   M  24.0   183.0     Russia   
282620      271109  Aleksandr Viktorovich Zyuzin   M  28.0   183.0     Russia   
282621      271110        Olga Igorevna Zyuzkova   F  33.0   171.0    Belarus   

        NOC        Games  Year  Season  ...                Sport  \
0       NOR   Tokyo 2020  2020  Summer  ...         Cycling Road   
1       ESP   Tokyo 2020  2020  Summer  ...  Artistic Gymnastics   
2       ITA   Tokyo 2020  2020  Summer  ...               Rowing   
3       ESP   Tokyo 2020  2020  Summer  ...           Basketball   
4       ESP   Tokyo 2020  2020  Summer  ...           Basketball   
...     ...          ...   ...     ...  ...                  ...   
282616  ARG  2000 Summer  2000  Summer  ...               Hockey   
282617  ARG  2004 Summer  2004  Summer  ...               Hockey   
282619  RUS  2000 Summer  2000  Summer  ...               Rowing   
282620  RUS  2004 Summer  2004  Summer  ...               Rowing   
282621  BLR  2016 Summer  2016  Summer  ...           Basketball   

               Medal                                   Event        ID  \
0       Pas_medaille                                     NaN       NaN   
1       Pas_medaille                                     NaN       NaN   
2       Pas_medaille                                     NaN       NaN   
3       Pas_medaille                                     NaN       NaN   
4       Pas_medaille                                     NaN       NaN   
...              ...                                     ...       ...   
282616  Pas_medaille                     Hockey Men's Hockey  135565.0   
282617  Pas_medaille                     Hockey Men's Hockey  135565.0   
282619  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282620  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282621  Pas_medaille           Basketball Women's Basketball  135568.0   

        Weight  Country_org   Population  PIB/habitant  Score_medaille  \
0          NaN        Japan    5195921.0  74400.369777               0   
1          NaN        Japan   46418269.0  25831.582305               0   
2         81.0        Japan   60802085.0  29957.804315               0   
3         88.0        Japan   46418269.0  25831.582305               0   
4         88.0        Japan   46418269.0  25831.582305               0   
...        ...          ...          ...           ...             ...   
282616    76.0    Australia   43416755.0  13431.878340               0   
282617    76.0       Greece   43416755.0  13431.878340               0   
282619    72.0    Australia  144096812.0   9092.580536               0   
282620    72.0       Greece  144096812.0   9092.580536               0   
282621    69.0       Brazil    9513000.0   5740.456495               0   

          Country  
0          Norway  
1           Spain  
2           Italy  
3           Spain  
4           Spain  
...           ...  
282616  Argentina  
282617  Argentina  
282619     Russia  
282620     Russia  
282621    Belarus  

[78790 rows x 21 columns]

In [53] used -5.7 MiB RAM in 0.18s (system mean cpu 39%, single max cpu 100%), peaked 5.9 MiB above final usage, current RAM usage now 453.8 MiB


## Moyenne par pays pour les sports qu'on ne connait pas 


In [54]:
table_pays_1=df_complete[df_complete['Weight'].isna()==False].groupby(['Country'])['Weight'].mean().apply(m.ceil).reset_index()

liste_pays_1=table_pays_1['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_1=table_pays_1['Weight'].tolist()



dictio_pays_poids_1=dict(zip(liste_pays_1,liste_poids_1))
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Country'].map(dictio_pays_poids_1))

In [54] used 0.2 MiB RAM in 0.14s (system mean cpu 13%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 454.0 MiB


In [55]:
table_pays_2=df_complete[df_complete['Height'].isna()==False].groupby(['Country'])['Height'].mean().apply(m.ceil).reset_index()

liste_pays_2=table_pays_2['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_2=table_pays_2['Height'].tolist()



dictio_pays_poids_2=dict(zip(liste_pays_2,liste_poids_2))
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Country'].map(dictio_pays_poids_2))

#df_complete.isna().sum()

In [55] used 5.0 MiB RAM in 0.14s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 459.0 MiB


In [56]:
#df_complete.to_csv('data/df_merge_FLO.csv', index=False)

In [56] used 0.0 MiB RAM in 0.10s (system mean cpu 9%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 459.0 MiB


## Web Scrapping FLo 

In [57]:
from bs4 import BeautifulSoup
import requests

In [57] used 0.3 MiB RAM in 0.30s (system mean cpu 13%, single max cpu 100%), peaked 5.0 MiB above final usage, current RAM usage now 459.4 MiB


In [58]:
url='https://www.topendsports.com/events/summer/hosts/list.htm'
response=requests.get(url)

In [58] used -18.5 MiB RAM in 1.25s (system mean cpu 7%, single max cpu 100%), peaked 21.3 MiB above final usage, current RAM usage now 440.9 MiB


In [59]:
url_2='https://en.wikipedia.org/wiki/List_of_Olympic_Games_host_cities'

In [59] used 0.0 MiB RAM in 0.11s (system mean cpu 11%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 440.9 MiB


In [60]:
response_2=requests.get(url_2)


In [60] used 0.9 MiB RAM in 0.94s (system mean cpu 5%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 441.8 MiB


In [61]:
soup_2=BeautifulSoup(response_2.text,'html.parser')


In [61] used 0.5 MiB RAM in 0.25s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 442.3 MiB


## On scrapp les pays organisateurs 

In [62]:
L_pays=[]
L_annees=[]
elements=soup_2.findAll('span',class_='datasortkey')

for element in elements:
    pays=element.find('a').text
    L_pays.append(pays)

L_pays=L_pays[42:58]




In [62] used 0.0 MiB RAM in 0.12s (system mean cpu 9%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 442.3 MiB


In [63]:
L_annees_2=[2000 + i  for i in range(0,34,2)]
L_annees_2.pop(-2)

L_annees_2

[2000,
 2002,
 2004,
 2006,
 2008,
 2010,
 2012,
 2014,
 2016,
 2018,
 2020,
 2022,
 2024,
 2026,
 2028,
 2032]

In [63] used -0.0 MiB RAM in 0.11s (system mean cpu 17%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 442.3 MiB


In [64]:
L_pays_annee=[]
for i in range(len(L_pays)):
    L_pays_annee.append((L_pays[i],L_annees_2[i]))

df_hote=pd.DataFrame(L_pays_annee,columns=['Host_country','Year'])

df_hote




Host_country  Year
0        Australia  2000
1    United States  2002
2           Greece  2004
3            Italy  2006
4            China  2008
5           Canada  2010
6   United Kingdom  2012
7           Russia  2014
8           Brazil  2016
9      South Korea  2018
10           Japan  2020
11           China  2022
12          France  2024
13           Italy  2026
14   United States  2028
15       Australia  2032

In [64] used 0.1 MiB RAM in 0.12s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 442.4 MiB


In [65]:
df_complete

Unnamed: 0                          Name Sex   Age  Height       Team  \
0                0               AALERUD Katrine   F  27.0   175.0     Norway   
1                1                   ABAD Nestor   M  28.0   165.0      Spain   
2                2             ABAGNALE Giovanni   M  26.0   198.0      Italy   
3                3                ABALDE Alberto   M  26.0   200.0      Spain   
4                4                 ABALDE Tamara   F  32.0   192.0      Spain   
...            ...                           ...  ..   ...     ...        ...   
282616      271105      Fernando scar Zylberberg   M  23.0   168.0  Argentina   
282617      271106      Fernando scar Zylberberg   M  27.0   168.0  Argentina   
282619      271108  Aleksandr Viktorovich Zyuzin   M  24.0   183.0     Russia   
282620      271109  Aleksandr Viktorovich Zyuzin   M  28.0   183.0     Russia   
282621      271110        Olga Igorevna Zyuzkova   F  33.0   171.0    Belarus   

        NOC        Games  Year  Season  ...                Sport  \
0       NOR   Tokyo 2020  2020  Summer  ...         Cycling Road   
1       ESP   Tokyo 2020  2020  Summer  ...  Artistic Gymnastics   
2       ITA   Tokyo 2020  2020  Summer  ...               Rowing   
3       ESP   Tokyo 2020  2020  Summer  ...           Basketball   
4       ESP   Tokyo 2020  2020  Summer  ...           Basketball   
...     ...          ...   ...     ...  ...                  ...   
282616  ARG  2000 Summer  2000  Summer  ...               Hockey   
282617  ARG  2004 Summer  2004  Summer  ...               Hockey   
282619  RUS  2000 Summer  2000  Summer  ...               Rowing   
282620  RUS  2004 Summer  2004  Summer  ...               Rowing   
282621  BLR  2016 Summer  2016  Summer  ...           Basketball   

               Medal                                   Event        ID  \
0       Pas_medaille                                     NaN       NaN   
1       Pas_medaille                                     NaN       NaN   
2       Pas_medaille                                     NaN       NaN   
3       Pas_medaille                                     NaN       NaN   
4       Pas_medaille                                     NaN       NaN   
...              ...                                     ...       ...   
282616  Pas_medaille                     Hockey Men's Hockey  135565.0   
282617  Pas_medaille                     Hockey Men's Hockey  135565.0   
282619  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282620  Pas_medaille  Rowing Men's Lightweight Coxless Fours  135567.0   
282621  Pas_medaille           Basketball Women's Basketball  135568.0   

        Weight  Country_org   Population  PIB/habitant  Score_medaille  \
0         74.0        Japan    5195921.0  74400.369777               0   
1         71.0        Japan   46418269.0  25831.582305               0   
2         81.0        Japan   60802085.0  29957.804315               0   
3         88.0        Japan   46418269.0  25831.582305               0   
4         88.0        Japan   46418269.0  25831.582305               0   
...        ...          ...          ...           ...             ...   
282616    76.0    Australia   43416755.0  13431.878340               0   
282617    76.0       Greece   43416755.0  13431.878340               0   
282619    72.0    Australia  144096812.0   9092.580536               0   
282620    72.0       Greece  144096812.0   9092.580536               0   
282621    69.0       Brazil    9513000.0   5740.456495               0   

          Country  
0          Norway  
1           Spain  
2           Italy  
3           Spain  
4           Spain  
...           ...  
282616  Argentina  
282617  Argentina  
282619     Russia  
282620     Russia  
282621    Belarus  

[78790 rows x 21 columns]

In [65] used 8.4 MiB RAM in 0.17s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 450.8 MiB


In [66]:
df_complete=pd.merge(df_complete,df_hote,on='Year',how='left')

df_complete

Unnamed: 0                          Name Sex   Age  Height       Team  \
0               0               AALERUD Katrine   F  27.0   175.0     Norway   
1               1                   ABAD Nestor   M  28.0   165.0      Spain   
2               2             ABAGNALE Giovanni   M  26.0   198.0      Italy   
3               3                ABALDE Alberto   M  26.0   200.0      Spain   
4               4                 ABALDE Tamara   F  32.0   192.0      Spain   
...           ...                           ...  ..   ...     ...        ...   
78785      271105      Fernando scar Zylberberg   M  23.0   168.0  Argentina   
78786      271106      Fernando scar Zylberberg   M  27.0   168.0  Argentina   
78787      271108  Aleksandr Viktorovich Zyuzin   M  24.0   183.0     Russia   
78788      271109  Aleksandr Viktorovich Zyuzin   M  28.0   183.0     Russia   
78789      271110        Olga Igorevna Zyuzkova   F  33.0   171.0    Belarus   

       NOC        Games  Year  Season  ...         Medal  \
0      NOR   Tokyo 2020  2020  Summer  ...  Pas_medaille   
1      ESP   Tokyo 2020  2020  Summer  ...  Pas_medaille   
2      ITA   Tokyo 2020  2020  Summer  ...  Pas_medaille   
3      ESP   Tokyo 2020  2020  Summer  ...  Pas_medaille   
4      ESP   Tokyo 2020  2020  Summer  ...  Pas_medaille   
...    ...          ...   ...     ...  ...           ...   
78785  ARG  2000 Summer  2000  Summer  ...  Pas_medaille   
78786  ARG  2004 Summer  2004  Summer  ...  Pas_medaille   
78787  RUS  2000 Summer  2000  Summer  ...  Pas_medaille   
78788  RUS  2004 Summer  2004  Summer  ...  Pas_medaille   
78789  BLR  2016 Summer  2016  Summer  ...  Pas_medaille   

                                        Event        ID Weight  Country_org  \
0                                         NaN       NaN   74.0        Japan   
1                                         NaN       NaN   71.0        Japan   
2                                         NaN       NaN   81.0        Japan   
3                                         NaN       NaN   88.0        Japan   
4                                         NaN       NaN   88.0        Japan   
...                                       ...       ...    ...          ...   
78785                     Hockey Men's Hockey  135565.0   76.0    Australia   
78786                     Hockey Men's Hockey  135565.0   76.0       Greece   
78787  Rowing Men's Lightweight Coxless Fours  135567.0   72.0    Australia   
78788  Rowing Men's Lightweight Coxless Fours  135567.0   72.0       Greece   
78789           Basketball Women's Basketball  135568.0   69.0       Brazil   

        Population  PIB/habitant  Score_medaille    Country  Host_country  
0        5195921.0  74400.369777               0     Norway         Japan  
1       46418269.0  25831.582305               0      Spain         Japan  
2       60802085.0  29957.804315               0      Italy         Japan  
3       46418269.0  25831.582305               0      Spain         Japan  
4       46418269.0  25831.582305               0      Spain         Japan  
...            ...           ...             ...        ...           ...  
78785   43416755.0  13431.878340               0  Argentina     Australia  
78786   43416755.0  13431.878340               0  Argentina        Greece  
78787  144096812.0   9092.580536               0     Russia     Australia  
78788  144096812.0   9092.580536               0     Russia        Greece  
78789    9513000.0   5740.456495               0    Belarus        Brazil  

[78790 rows x 22 columns]

In [66] used 0.1 MiB RAM in 0.17s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 451.0 MiB


In [67]:
compare_col=lambda x : 1 if x['Country']==x['Host_country'] else 0 




In [67] used -0.0 MiB RAM in 0.10s (system mean cpu 8%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 450.9 MiB


In [68]:
df_complete['Participation_own_country']=df_complete.apply(compare_col,axis=1)

df_complete[df_complete['Participation_own_country']==1]

Unnamed: 0               Name Sex   Age  Height    Team  NOC  \
38             45      ABE Takatoshi   M  30.0   177.0   Japan  JPN   
79             91      ADACHI Kazuya   M  31.0   171.0   Japan  JPN   
80             92       ADACHI Seiya   M  26.0   172.0   Japan  JPN   
109           126        AGARIE Yuto   M  28.0   184.0   Japan  JPN   
148           178       AIZAWA Akira   M  24.0   177.0   Japan  JPN   
...           ...                ...  ..   ...     ...     ...  ...   
78604      270431            Zou Kai   M  20.0   158.0   China  CHN   
78605      270432            Zou Kai   M  20.0   158.0   China  CHN   
78606      270433            Zou Kai   M  20.0   158.0   China  CHN   
78612      270439        Zou Shiming   M  27.0   165.0   China  CHN   
78782      271099  Stavroula Zygouri   F  36.0   171.0  Greece  GRE   

             Games  Year  Season  ...  \
38      Tokyo 2020  2020  Summer  ...   
79      Tokyo 2020  2020  Summer  ...   
80      Tokyo 2020  2020  Summer  ...   
109     Tokyo 2020  2020  Summer  ...   
148     Tokyo 2020  2020  Summer  ...   
...            ...   ...     ...  ...   
78604  2008 Summer  2008  Summer  ...   
78605  2008 Summer  2008  Summer  ...   
78606  2008 Summer  2008  Summer  ...   
78612  2008 Summer  2008  Summer  ...   
78782  2004 Summer  2004  Summer  ...   

                                           Event        ID Weight  \
38                                           NaN       NaN   70.0   
79                                           NaN       NaN   66.0   
80                                           NaN       NaN   85.0   
109                                          NaN       NaN   82.0   
148                                          NaN       NaN   70.0   
...                                          ...       ...    ...   
78604           Gymnastics Men's Team All-Around  135269.0   55.0   
78605            Gymnastics Men's Floor Exercise  135269.0   55.0   
78606            Gymnastics Men's Horizontal Bar  135269.0   55.0   
78612               Boxing Men's Light-Flyweight  135270.0   49.0   
78782  Wrestling Women's Middleweight, Freestyle  135560.0   63.0   

                      Country_org    Population  PIB/habitant Score_medaille  \
38                          Japan  1.269585e+08  32477.215145              0   
79                          Japan  1.269585e+08  32477.215145              0   
80                          Japan  1.269585e+08  32477.215145              0   
109                         Japan  1.269585e+08  32477.215145              0   
148                         Japan  1.269585e+08  32477.215145              0   
...                           ...           ...           ...            ...   
78604  People's Republic of China  1.371220e+09   8027.683810              3   
78605  People's Republic of China  1.371220e+09   8027.683810              3   
78606  People's Republic of China  1.371220e+09   8027.683810              3   
78612  People's Republic of China  1.371220e+09   8027.683810              3   
78782                      Greece  1.082373e+07  18002.230578              0   

       Country  Host_country  Participation_own_country  
38       Japan         Japan                          1  
79       Japan         Japan                          1  
80       Japan         Japan                          1  
109      Japan         Japan                          1  
148      Japan         Japan                          1  
...        ...           ...                        ...  
78604    China         China                          1  
78605    China         China                          1  
78606    China         China                          1  
78612    China         China                          1  
78782   Greece        Greece                          1  

[3899 rows x 23 columns]

In [68] used -15.3 MiB RAM in 0.55s (system mean cpu 17%, single max cpu 100%), peaked 15.5 MiB above final usage, current RAM usage now 435.6 MiB


In [69]:
table_sports=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sports

Sport   Name
0          3x3 Basketball     56
1                 Archery   1139
2     Artistic Gymnastics    191
3       Artistic Swimming    100
4               Athletics  13711
5               Badminton   1135
6                Baseball    574
7       Baseball/Softball    234
8              Basketball   1715
9        Beach Volleyball    570
10                 Boxing   1711
11           Canoe Slalom     80
12           Canoe Sprint    243
13               Canoeing   2177
14                Cycling   3192
15  Cycling BMX Freestyle     16
16     Cycling BMX Racing     42
17  Cycling Mountain Bike     74
18           Cycling Road    189
19          Cycling Track    224
20                 Diving   1114
21             Equestrian    232
22          Equestrianism   1711
23                Fencing   1937
24               Football   2807
25                   Golf    240
26             Gymnastics   5000
27               Handball   2034
28                 Hockey   2272
29                   Judo   2330
30                 Karate     79
31      Marathon Swimming     47
32      Modern Pentathlon    391
33    Rhythmic Gymnastics    543
34                 Rowing   3282
35           Rugby Sevens    598
36                Sailing   2301
37               Shooting   3190
38          Skateboarding     79
39               Softball    358
40         Sport Climbing     36
41                Surfing     40
42               Swimming   9124
43  Synchronized Swimming    586
44           Table Tennis   1425
45              Taekwondo    728
46                 Tennis   1508
47  Trampoline Gymnastics     28
48           Trampolining    152
49              Triathlon    637
50             Volleyball   1677
51             Water Polo   1530
52          Weightlifting   1446
53              Wrestling   1955

In [69] used 0.0 MiB RAM in 0.12s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 435.7 MiB


## Dictionnaire group sport

On va essayer de scrapper les sports individuels et les collectifs 


In [70]:
url_sports='https://www.britannica.com/topic/list-of-sports-2038581#ref330452'

In [70] used -0.0 MiB RAM in 0.11s (system mean cpu 8%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 435.6 MiB


In [71]:
response_sp=requests.get(url_sports)

In [71] used -1.6 MiB RAM in 0.88s (system mean cpu 9%, single max cpu 67%), peaked 1.8 MiB above final usage, current RAM usage now 434.0 MiB


In [72]:
soup_sports=BeautifulSoup(response_sp.text,'html.parser')


In [72] used 49.2 MiB RAM in 0.24s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 483.2 MiB


In [73]:
sports=soup_sports.findAll('div')
L_sports_indiv=[]
for sport in sports:
    a=sport.find('a',class_='md-crosslink')
    if a is not None:
        L_sports_indiv.append(a.text)
   
    
L_sports_collec=[]

sports_co=soup_sports.findAll('section',id='ref330452')

for sport in sports_co:
    b=sport.find('a',class_='md-crosslink')
    print(b)
   
        
#sports

<a class="md-crosslink" data-show-preview="true" href="https://www.britannica.com/sports/association-croquet">association croquet</a>
In [73] used -0.0 MiB RAM in 0.13s (system mean cpu 21%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 483.2 MiB


## Pays passé communiste 

In [74]:
url_soviet='https://en.wikipedia.org/wiki/Post-Soviet_states'

In [74] used -0.0 MiB RAM in 0.10s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 483.1 MiB


In [75]:
response_soviet=requests.get(url_soviet)

In [75] used 0.4 MiB RAM in 0.19s (system mean cpu 9%, single max cpu 57%), peaked 0.0 MiB above final usage, current RAM usage now 483.6 MiB


In [76]:
soup_soviet=BeautifulSoup(response_soviet.text,'html.parser')

In [76] used 0.1 MiB RAM in 0.34s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 483.6 MiB


In [77]:
L_pays_soviet=[]

pays_soviet=soup_soviet.findAll('b')
#print(pays_soviet)
for pays in pays_soviet:
    b=pays.find('a')
    if b is not None:
        L_pays_soviet.append(b.text)
L_pays_soviet=L_pays_soviet[:15]

L_pays_soviet

['Kazakhstan',
 'Kyrgyzstan',
 'Tajikistan',
 'Turkmenistan',
 'Uzbekistan',
 'Belarus',
 'Moldova',
 'Russia',
 'Ukraine',
 'Estonia',
 'Latvia',
 'Lithuania',
 'Armenia',
 'Azerbaijan',
 'Georgia']

In [77] used 0.0 MiB RAM in 0.12s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 483.6 MiB


In [78]:
complete_urss_regime=lambda x : 1 if x in L_pays_soviet else 0

In [78] used -8.6 MiB RAM in 0.11s (system mean cpu 11%, single max cpu 75%), peaked 8.6 MiB above final usage, current RAM usage now 475.1 MiB


In [79]:
df_complete['Soviet_past']=df_complete['Country'].apply(complete_urss_regime)



In [79] used -0.0 MiB RAM in 0.17s (system mean cpu 12%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 475.1 MiB


In [80]:
df_complete.isna().sum()

Unnamed: 0                       0
Name                             0
Sex                              0
Age                              0
Height                           0
Team                             0
NOC                              0
Games                            0
Year                             0
Season                           0
City                             0
Sport                            0
Medal                            0
Event                         9093
ID                           11328
Weight                           0
Country_org                      0
Population                    2609
PIB/habitant                  5315
Score_medaille                   0
Country                          0
Host_country                     0
Participation_own_country        0
Soviet_past                      0
dtype: int64

In [80] used 0.1 MiB RAM in 0.17s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.1 MiB


In [81]:
df_pop_na=df_complete[df_complete['Population'].isna()]

#df_pop_na.groupby('Country')['Name'].count().reset_index()

In [81] used 0.1 MiB RAM in 0.11s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.3 MiB


## Scrap Population et GPA 

In [82]:
import re

In [82] used 0.0 MiB RAM in 0.10s (system mean cpu 14%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 475.3 MiB


In [83]:
url_pop_gpa='https://www.worldometers.info/gdp/gdp-by-country/'

In [83] used 0.0 MiB RAM in 0.11s (system mean cpu 9%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.3 MiB


In [84]:
response_pop_gpa=requests.get(url_pop_gpa)


In [84] used 0.1 MiB RAM in 0.56s (system mean cpu 8%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.4 MiB


In [85]:
soup_pop_gpa=BeautifulSoup(response_pop_gpa.text,'html.parser')


In [85] used 0.0 MiB RAM in 0.19s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.4 MiB


In [86]:
L_pays=[]

all_countries=soup_pop_gpa.findAll('td')

for country in all_countries:
    a=country.find('a')
    if a is not None:
        L_pays.append(a.text)
len(L_pays)

177

In [86] used 0.0 MiB RAM in 0.12s (system mean cpu 13%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.4 MiB


In [87]:
#b=re.findall(r'\d+\,?\d*|\d+\.\d+', info.text)

In [87] used 0.1 MiB RAM in 0.11s (system mean cpu 9%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.5 MiB


In [88]:
all_info=soup_pop_gpa.findAll('td',attrs={'style':'font-weight: bold; text-align:right;'})
#print(all_info)
numbers=[]
for i,info in enumerate(all_info):
    b=re.findall(r'\d+\b', info.text)
    b=''.join(b)
    numbers.append(b)
   
 
    #b=[number.replace(',','.') for number in b]
#print(len(numbers))
population=[]
gpd_per_habitant=[]
for i in range(3,len(numbers),6):
    population.append(int(numbers[i]))
for i in range(4,len(numbers),6):
    gpd_per_habitant.append(int(numbers[i]))
    
print(len(population))
print(len(gpd_per_habitant))

177
177
In [88] used 0.0 MiB RAM in 0.13s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 475.5 MiB


In [106]:
df_pop_gpd=pd.DataFrame({'Country':L_pays,'Population':population,'PIB/habitant':gpd_per_habitant})

df_pop_gpd['Country']=df_pop_gpd['Country'].replace('Trinidad and Tobago','Trinidad')
df_pop_gpd['Country']=df_pop_gpd['Country'].replace('State of Palestine','Palestine, Occupied Territories')
df_pop_gpd['Country']=df_pop_gpd['Country'].replace('Singapore','Singapour')

#new_row=pd.DataFrame({'Country':'Taiwan','Population':23923276,'PIB/habitant':44460})
#df_pop_gpd=df_pop_gpd.append(new_row,ignore_index=True)

df_pop_gpd.loc[1]=['Taiwan',23923276,44460]
df_pop_gpd.loc[2]=['Cook Islands',17044,13558]
df_pop_gpd.loc[3]=['Eritrea',3684000,1067]
df_pop_gpd.loc[4]=['Kosovo',1760000,5340]
df_pop_gpd.loc[5]=['Liban',5425542,4136]
df_pop_gpd.loc[6]=['Netherlands Antilles*',27148,11400]
df_pop_gpd.loc[7]=['South Sudan',11000000,467]








df_pop_gpd[df_pop_gpd['Country']=='Iran']


Country  Population  PIB/habitant
39    Iran    88550570          4388

In [106] used -4.6 MiB RAM in 0.13s (system mean cpu 16%, single max cpu 28%), peaked 0.0 MiB above final usage, current RAM usage now 27.0 MiB


In [90]:
df_pop_gpd.head(20)

Country  Population  PIB/habitant
0           United States   338289857         75269
1                  Taiwan    23923276         44460
2            Cook Islands       17044         13558
3                 Eritrea     3684000          1067
4                  Kosovo     1760000          5340
5                   Liban     5425542          4136
6   Netherlands Antilles*       27148         11400
7             South Sudan    11000000           467
8                  Canada    38454327         55646
9                   Italy    59037474         34053
10                 Brazil   215313498          8918
11              Australia    26177413         64003
12            South Korea    51815810         32138
13                 Mexico   127504125         11091
14                  Spain    47558630         29385
15              Indonesia   275501339          4788
16           Saudi Arabia    36408820         30436
17            Netherlands    17564014         56429
18                 Turkey    85341241         10616
19            Switzerland     8740472         92410

In [90] used -0.0 MiB RAM in 0.12s (system mean cpu 9%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 475.5 MiB


## Dictionnaire pour remplir les valeurs NaN de Pop et PIB sans changer les valeurs de pop et PIB per habitant précédentes

In [94]:
dictio_pop_gpd=df_pop_gpd.to_dict()

dictio_pop_gpd

{'Country': {0: 'United States',
  1: 'Taiwan',
  2: 'Cook Islands',
  3: 'Eritrea',
  4: 'Kosovo',
  5: 'Liban',
  6: 'Netherlands Antilles*',
  7: 'South Sudan',
  8: 'Canada',
  9: 'Italy',
  10: 'Brazil',
  11: 'Australia',
  12: 'South Korea',
  13: 'Mexico',
  14: 'Spain',
  15: 'Indonesia',
  16: 'Saudi Arabia',
  17: 'Netherlands',
  18: 'Turkey',
  19: 'Switzerland',
  20: 'Poland',
  21: 'Argentina',
  22: 'Sweden',
  23: 'Norway',
  24: 'Belgium',
  25: 'Ireland',
  26: 'Israel',
  27: 'United Arab Emirates',
  28: 'Thailand',
  29: 'Nigeria',
  30: 'Egypt',
  31: 'Austria',
  32: 'Singapour',
  33: 'Bangladesh',
  34: 'Vietnam',
  35: 'Malaysia',
  36: 'South Africa',
  37: 'Philippines',
  38: 'Denmark',
  39: 'Iran',
  40: 'Pakistan',
  41: 'Hong Kong',
  42: 'Colombia',
  43: 'Romania',
  44: 'Chile',
  45: 'Czech Republic (Czechia)',
  46: 'Finland',
  47: 'Iraq',
  48: 'Portugal',
  49: 'New Zealand',
  50: 'Peru',
  51: 'Qatar',
  52: 'Kazakhstan',
  53: 'Greece',
  5

In [94] used -426.1 MiB RAM in 0.14s (system mean cpu 23%, single max cpu 41%), peaked 0.0 MiB above final usage, current RAM usage now 22.5 MiB


In [100]:
df_test=df_complete.copy()



In [100] used 10.6 MiB RAM in 0.18s (system mean cpu 29%, single max cpu 46%), peaked 0.0 MiB above final usage, current RAM usage now 90.4 MiB


In [101]:
df_test.head(20)

Unnamed: 0                     Name Sex   Age  Height  \
0            0          AALERUD Katrine   F  27.0   175.0   
1            1              ABAD Nestor   M  28.0   165.0   
2            2        ABAGNALE Giovanni   M  26.0   198.0   
3            3           ABALDE Alberto   M  26.0   200.0   
4            4            ABALDE Tamara   F  32.0   192.0   
5            6             ABAROA Cesar   M  25.0   179.0   
6            7            ABASS Abobakr   M  23.0   181.0   
7            8       ABBAGNALE Vincenzo   M  28.0   187.0   
8            9         ABBASALI Hamideh   F  31.0   181.0   
9           10            ABBASOV Islam   M  25.0   174.0   
10          11             ABBINGH Lois   F  29.0   177.0   
11          12              ABBOT Emily   F  24.0   160.0   
12          14  ABDALLA Abubaker Haydar   M  25.0   177.0   
13          15      ABDALRASOOL Mohamed   M  28.0   174.0   
14          16        ABDEL LATIF Radwa   F  32.0   173.0   
15          17         ABDEL RAZEK Samy   M  41.0   173.0   
16          18         ABDELAAL Mohamed   M  31.0   174.0   
17          19        ABDELAZIZ Abdalla   M  22.0   178.0   
18          20          ABDELAZIZ Farah   F  29.0   172.0   
19          22     ABDELMAWGOUD Mohamed   M  27.0   174.0   

                        Team  NOC       Games  Year  Season  ...  ID Weight  \
0                     Norway  NOR  Tokyo 2020  2020  Summer  ... NaN   74.0   
1                      Spain  ESP  Tokyo 2020  2020  Summer  ... NaN   71.0   
2                      Italy  ITA  Tokyo 2020  2020  Summer  ... NaN   81.0   
3                      Spain  ESP  Tokyo 2020  2020  Summer  ... NaN   88.0   
4                      Spain  ESP  Tokyo 2020  2020  Summer  ... NaN   88.0   
5                      Chile  CHI  Tokyo 2020  2020  Summer  ... NaN   81.0   
6                      Sudan  SUD  Tokyo 2020  2020  Summer  ... NaN   73.0   
7                      Italy  ITA  Tokyo 2020  2020  Summer  ... NaN   81.0   
8   Islamic Republic of Iran  IRI  Tokyo 2020  2020  Summer  ... NaN   81.0   
9                 Azerbaijan  AZE  Tokyo 2020  2020  Summer  ... NaN   77.0   
10               Netherlands  NED  Tokyo 2020  2020  Summer  ... NaN   82.0   
11                 Australia  AUS  Tokyo 2020  2020  Summer  ... NaN   50.0   
12                     Qatar  QAT  Tokyo 2020  2020  Summer  ... NaN   70.0   
13                     Sudan  SUD  Tokyo 2020  2020  Summer  ... NaN   79.0   
14                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   74.0   
15                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   74.0   
16                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   79.0   
17                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   75.0   
18                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   66.0   
19                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   79.0   

   Country_org  Population  PIB/habitant  Score_medaille      Country  \
0        Japan   5195921.0  74400.369777               0       Norway   
1        Japan  46418269.0  25831.582305               0        Spain   
2        Japan  60802085.0  29957.804315               0        Italy   
3        Japan  46418269.0  25831.582305               0        Spain   
4        Japan  46418269.0  25831.582305               0        Spain   
5        Japan  17948141.0  13416.230039               0        Chile   
6        Japan  40234882.0   2414.723601               0        Sudan   
7        Japan  60802085.0  29957.804315               0        Italy   
8        Japan  79109272.0           NaN               0         Iran   
9        Japan   9651349.0   5496.344640               0   Azerbaijan   
10       Japan  16936520.0  44299.768085               0  Netherlands   
11       Japan  23781169.0  56310.962993               0    Australia   
12       Japan   2235355.0  73653.394435               0        Qatar   
13       Japan  40234882.0   2414.723601

In [101] used -63.8 MiB RAM in 0.16s (system mean cpu 20%, single max cpu 29%), peaked 0.0 MiB above final usage, current RAM usage now 26.7 MiB


In [105]:
df_test['PIB/habitant']=df_test['PIB/habitant'].fillna(df_pop_gpd['PIB/habitant'])

df_test.head(20)

Unnamed: 0                     Name Sex   Age  Height  \
0            0          AALERUD Katrine   F  27.0   175.0   
1            1              ABAD Nestor   M  28.0   165.0   
2            2        ABAGNALE Giovanni   M  26.0   198.0   
3            3           ABALDE Alberto   M  26.0   200.0   
4            4            ABALDE Tamara   F  32.0   192.0   
5            6             ABAROA Cesar   M  25.0   179.0   
6            7            ABASS Abobakr   M  23.0   181.0   
7            8       ABBAGNALE Vincenzo   M  28.0   187.0   
8            9         ABBASALI Hamideh   F  31.0   181.0   
9           10            ABBASOV Islam   M  25.0   174.0   
10          11             ABBINGH Lois   F  29.0   177.0   
11          12              ABBOT Emily   F  24.0   160.0   
12          14  ABDALLA Abubaker Haydar   M  25.0   177.0   
13          15      ABDALRASOOL Mohamed   M  28.0   174.0   
14          16        ABDEL LATIF Radwa   F  32.0   173.0   
15          17         ABDEL RAZEK Samy   M  41.0   173.0   
16          18         ABDELAAL Mohamed   M  31.0   174.0   
17          19        ABDELAZIZ Abdalla   M  22.0   178.0   
18          20          ABDELAZIZ Farah   F  29.0   172.0   
19          22     ABDELMAWGOUD Mohamed   M  27.0   174.0   

                        Team  NOC       Games  Year  Season  ...  ID Weight  \
0                     Norway  NOR  Tokyo 2020  2020  Summer  ... NaN   74.0   
1                      Spain  ESP  Tokyo 2020  2020  Summer  ... NaN   71.0   
2                      Italy  ITA  Tokyo 2020  2020  Summer  ... NaN   81.0   
3                      Spain  ESP  Tokyo 2020  2020  Summer  ... NaN   88.0   
4                      Spain  ESP  Tokyo 2020  2020  Summer  ... NaN   88.0   
5                      Chile  CHI  Tokyo 2020  2020  Summer  ... NaN   81.0   
6                      Sudan  SUD  Tokyo 2020  2020  Summer  ... NaN   73.0   
7                      Italy  ITA  Tokyo 2020  2020  Summer  ... NaN   81.0   
8   Islamic Republic of Iran  IRI  Tokyo 2020  2020  Summer  ... NaN   81.0   
9                 Azerbaijan  AZE  Tokyo 2020  2020  Summer  ... NaN   77.0   
10               Netherlands  NED  Tokyo 2020  2020  Summer  ... NaN   82.0   
11                 Australia  AUS  Tokyo 2020  2020  Summer  ... NaN   50.0   
12                     Qatar  QAT  Tokyo 2020  2020  Summer  ... NaN   70.0   
13                     Sudan  SUD  Tokyo 2020  2020  Summer  ... NaN   79.0   
14                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   74.0   
15                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   74.0   
16                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   79.0   
17                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   75.0   
18                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   66.0   
19                     Egypt  EGY  Tokyo 2020  2020  Summer  ... NaN   79.0   

   Country_org  Population  PIB/habitant  Score_medaille      Country  \
0        Japan   5195921.0  74400.369777               0       Norway   
1        Japan  46418269.0  25831.582305               0        Spain   
2        Japan  60802085.0  29957.804315               0        Italy   
3        Japan  46418269.0  25831.582305               0        Spain   
4        Japan  46418269.0  25831.582305               0        Spain   
5        Japan  17948141.0  13416.230039               0        Chile   
6        Japan  40234882.0   2414.723601               0        Sudan   
7        Japan  60802085.0  29957.804315               0        Italy   
8        Japan  79109272.0  55646.000000               0         Iran   
9        Japan   9651349.0   5496.344640               0   Azerbaijan   
10       Japan  16936520.0  44299.768085               0  Netherlands   
11       Japan  23781169.0  56310.962993               0    Australia   
12       Japan   2235355.0  73653.394435               0        Qatar   
13       Japan  40234882.0   2414.723601

In [105] used -51.8 MiB RAM in 0.14s (system mean cpu 16%, single max cpu 26%), peaked 0.0 MiB above final usage, current RAM usage now 31.5 MiB


In [99]:
df_test=pd.merge(df_test,df_pop_gpd,on='Country',how='left')

df_test.head(20)

Unnamed: 0                     Name Sex   Age  Height  \
0            0          AALERUD Katrine   F  27.0   175.0   
1            1              ABAD Nestor   M  28.0   165.0   
2            2        ABAGNALE Giovanni   M  26.0   198.0   
3            3           ABALDE Alberto   M  26.0   200.0   
4            4            ABALDE Tamara   F  32.0   192.0   
5            6             ABAROA Cesar   M  25.0   179.0   
6            7            ABASS Abobakr   M  23.0   181.0   
7            8       ABBAGNALE Vincenzo   M  28.0   187.0   
8            9         ABBASALI Hamideh   F  31.0   181.0   
9           10            ABBASOV Islam   M  25.0   174.0   
10          11             ABBINGH Lois   F  29.0   177.0   
11          12              ABBOT Emily   F  24.0   160.0   
12          14  ABDALLA Abubaker Haydar   M  25.0   177.0   
13          15      ABDALRASOOL Mohamed   M  28.0   174.0   
14          16        ABDEL LATIF Radwa   F  32.0   173.0   
15          17         ABDEL RAZEK Samy   M  41.0   173.0   
16          18         ABDELAAL Mohamed   M  31.0   174.0   
17          19        ABDELAZIZ Abdalla   M  22.0   178.0   
18          20          ABDELAZIZ Farah   F  29.0   172.0   
19          22     ABDELMAWGOUD Mohamed   M  27.0   174.0   

                        Team  NOC       Games  Year  Season  ... Country_org  \
0                     Norway  NOR  Tokyo 2020  2020  Summer  ...       Japan   
1                      Spain  ESP  Tokyo 2020  2020  Summer  ...       Japan   
2                      Italy  ITA  Tokyo 2020  2020  Summer  ...       Japan   
3                      Spain  ESP  Tokyo 2020  2020  Summer  ...       Japan   
4                      Spain  ESP  Tokyo 2020  2020  Summer  ...       Japan   
5                      Chile  CHI  Tokyo 2020  2020  Summer  ...       Japan   
6                      Sudan  SUD  Tokyo 2020  2020  Summer  ...       Japan   
7                      Italy  ITA  Tokyo 2020  2020  Summer  ...       Japan   
8   Islamic Republic of Iran  IRI  Tokyo 2020  2020  Summer  ...       Japan   
9                 Azerbaijan  AZE  Tokyo 2020  2020  Summer  ...       Japan   
10               Netherlands  NED  Tokyo 2020  2020  Summer  ...       Japan   
11                 Australia  AUS  Tokyo 2020  2020  Summer  ...       Japan   
12                     Qatar  QAT  Tokyo 2020  2020  Summer  ...       Japan   
13                     Sudan  SUD  Tokyo 2020  2020  Summer  ...       Japan   
14                     Egypt  EGY  Tokyo 2020  2020  Summer  ...       Japan   
15                     Egypt  EGY  Tokyo 2020  2020  Summer  ...       Japan   
16                     Egypt  EGY  Tokyo 2020  2020  Summer  ...       Japan   
17                     Egypt  EGY  Tokyo 2020  2020  Summer  ...       Japan   
18                     Egypt  EGY  Tokyo 2020  2020  Summer  ...       Japan   
19                     Egypt  EGY  Tokyo 2020  2020  Summer  ...       Japan   

   Population_x PIB/habitant_x Score_medaille      Country  Host_country  \
0     5195921.0   74400.369777              0       Norway         Japan   
1    46418269.0   25831.582305              0        Spain         Japan   
2    60802085.0   29957.804315              0        Italy         Japan   
3    46418269.0   25831.582305              0        Spain         Japan   
4    46418269.0   25831.582305              0        Spain         Japan   
5    17948141.0   13416.230039              0        Chile         Japan   
6    40234882.0    2414.723601              0        Sudan         Japan   
7    60802085.0   29957.804315              0        Italy         Japan   
8    79109272.0            NaN              0         Iran         Japan   
9     9651349.0    5496.344640              0   Azerbaijan         Japan   
10   16936520.0   44299.768085              0  Netherlands         Japan   
11   23781169.0   56310.962993              0    Australia         Japan   
12    2235355.0   73653.394435              0        

In [99] used 53.8 MiB RAM in 0.26s (system mean cpu 61%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 79.8 MiB


In [92]:
df_pop_na[df_pop_na['Population_y'].isna()].groupby('Country')['Name'].count().reset_index()


Country  Name
0  Individual Olympic Athletes    17

In [92] used 0.0 MiB RAM in 0.11s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 476.0 MiB


In [93]:
df_pop_na[df_pop_na['Country']=='Netherlands Antilles*']

Unnamed: 0                                 Name Sex   Age  Height  \
904        49180                          Michel Daou   M  39.0   171.0   
907        49574                Rodion Rosko Davelaar   M  18.0   183.0   
944        57993             Caimin Christian Douglas   M  23.0   179.0   
1000       62940                 Philip Yousef Elhage   M  26.0   172.0   
1109       80756                      Geronimo Goeloe   M  22.0   186.0   
1179       96158                  Howard Xavier Hinds   M  22.0   184.0   
1180       96159                  Howard Xavier Hinds   M  22.0   184.0   
1236      100966                       Florencia Hunt   F  22.0   163.0   
1584      150778              Churandy Thomas Martina   M  20.0   180.0   
1585      150779              Churandy Thomas Martina   M  24.0   180.0   
1586      150780              Churandy Thomas Martina   M  24.0   180.0   
1601      156212                         Roland Melis   M  25.0   183.0   
2198      229440  Eduard Peter Nicolaas "Eddy" Stibbe   M  51.0   185.0   
2199      229441  Eduard Peter Nicolaas "Eddy" Stibbe   M  55.0   185.0   
2438      247922                      Cor van Aanholt   M  41.0   184.0   

                      Team  NOC        Games  Year  Season  ...  \
904   Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
907   Netherlands Antilles  AHO  2008 Summer  2008  Summer  ...   
944   Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
1000  Netherlands Antilles  AHO  2008 Summer  2008  Summer  ...   
1109  Netherlands Antilles  AHO  2004 Summer  2004  Summer  ...   
1179  Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
1180  Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
1236  Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
1584  Netherlands Antilles  AHO  2004 Summer  2004  Summer  ...   
1585  Netherlands Antilles  AHO  2008 Summer  2008  Summer  ...   
1586  Netherlands Antilles  AHO  2008 Summer  2008  Summer  ...   
1601  Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
2198  Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   
2199  Netherlands Antilles  AHO  2004 Summer  2004  Summer  ...   
2438  Netherlands Antilles  AHO  2000 Summer  2000  Summer  ...   

                     Country_org Population_x PIB/habitant_x Score_medaille  \
904                    Australia          NaN            NaN              0   
907   People's Republic of China          NaN            NaN              0   
944                    Australia          NaN            NaN              0   
1000  People's Republic of China          NaN            NaN              0   
1109                      Greece          NaN            NaN              0   
1179                   Australia          NaN            NaN              0   
1180                   Australia          NaN            NaN              0   
1236                   Australia          NaN            NaN              0   
1584                      Greece          NaN            NaN              0   
1585  People's Republic of China          NaN            NaN              0   
1586  People's Republic of China          NaN            NaN              0   
1601                   Australia          NaN            NaN              0   
2198                   Australia          NaN            NaN              0   
2199                      Greece          NaN            NaN              0   
2438                   Australia          NaN            NaN              0   

                    Country  Host_country Participation_own_country  \
904   Netherlands Antilles*     Australia                         0   
907   Netherlands Antilles*         China                         0   
944   Netherlands Antilles*     Australia                         0   
1000  Netherlands Antilles*         China                         0   
1109  Netherlands Antilles*        Greece                         0   
1179  Netherlands Antilles*     Australia

In [93] used -27.4 MiB RAM in 0.13s (system mean cpu 16%, single max cpu 100%), peaked 27.4 MiB above final usage, current RAM usage now 448.6 MiB
